### ISSUE TITLE
Update to C++17

### ISSUE THREAD

**Issue by frobnitzem** [qcscine/development-utils#1]
The following line seems to be where the C++14 standard is set.  However, sparrow uses C++17 (20?) features like std::transform that gcc will not import when the standard level is set to 14.

https://github.com/qcscine/development-utils/blob/413ce8aa2ccd1cdcdcd49366f493c90627fad5c7/cmake/ComponentSetup.cmake#L31

```
Sparrow/Embed/Common.h: In function 'std::string upper(std::string)':
>> 988     Sparrow/Embed/Common.h:107:8: error: 'transform' is not a member of 'std'
```

**Comment by weymutht** [comment]
Thanks for reporting this! We are actually in the process of migrating to a newer C++ standard. Interestingly enough, defining C++14 in `ComponentSetup.cmake` but using things like `std::transform` in Sparrow never caused any problems on our systems, which is why defining a newer C++ standard wasn't too high on our list. We hope to have this ready for the next release, though.

**Comment by reiher-research-group** [comment]
This is fixed in version 5.0.0 of development-utils, which has been released today.
